dbms/src/Interpreters/QueryPriorities.h

```diff
@@ -78,8 +82,16 @@ class QueryPriorities
                 return true;
 
             CurrentMetrics::Increment metric_increment{CurrentMetrics::QueryPreempted};
-            if (std::cv_status::timeout == condvar.wait_for(lock, timeout))
+            if (std::cv_status::timeout == condvar.wait_for(lock, cur_timeout))
                 return false;
+            else
+            {
+                /// False awakening, check and update time limit
```

Comment:
It's not necessarily the false awakening. It may awake due to convar being notified because of some query was finished.